diff --git a/dist/components/input/input.d.ts b/dist/components/input/input.d.ts
index d109d9ea4dfc229ee0f8ea98d55fbe89d333d3af..49f5c545b7922e5752427f799926f71b8e864ee1 100644
--- a/dist/components/input/input.d.ts
+++ b/dist/components/input/input.d.ts
@@ -115,6 +115,10 @@ export default class WaInput extends WebAwesomeFormAssociatedElement {
      * [this page on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) for available values.
      */
     autocomplete: string;
+    /**
+     * Sets the native input's `id` and the associated label `for`. Defaults to `"input"`.
+     */
+    inputId: string;
     /** Indicates that the input should receive focus on page load. */
     autofocus: boolean;
     /** Used to customize the label or icon of the Enter key on virtual keyboards. */
diff --git a/dist/components/input/input.js b/dist/components/input/input.js
index 1305a2e1e38f5e00bf3e4b6197ca85b332af84ef..12a3855a6dda00e285ca2f2ecc4e56a13c274db5 100644
--- a/dist/components/input/input.js
+++ b/dist/components/input/input.js
@@ -51,6 +51,7 @@ let WaInput = class extends WebAwesomeFormAssociatedElement {
     this.spellcheck = true;
     this.withLabel = false;
     this.withHint = false;
+    this.inputId = "input";
   }
   static get validators() {
     return [...super.validators, MirrorValidator()];
@@ -184,7 +185,7 @@ let WaInput = class extends WebAwesomeFormAssociatedElement {
       label: true,
       "has-label": hasLabel
     })}
-        for="input"
+        for=${this.inputId || "input"}
         aria-hidden=${hasLabel ? "false" : "true"}
       >
         <slot name="label">${this.label}</slot>
@@ -195,7 +196,7 @@ let WaInput = class extends WebAwesomeFormAssociatedElement {
 
         <input
           part="input"
-          id="input"
+          id=${this.inputId || "input"}
           class="control"
           type=${this.type === "password" && this.passwordVisible ? "text" : this.type}
           title=${this.title}
@@ -368,6 +369,9 @@ __decorateClass([
 __decorateClass([
   property()
 ], WaInput.prototype, "autocomplete", 2);
+__decorateClass([
+  property({ attribute: "input-id" })
+], WaInput.prototype, "inputId", 2);
 __decorateClass([
   property({ type: Boolean })
 ], WaInput.prototype, "autofocus", 2);
