diff --git a/lib/cook-raw-quasi.js b/lib/cook-raw-quasi.js
index 3ea8fa7be8e357c1066d7417caeeecd841415208..6bf04ab0bed8897b5ff2898ca835867aec5cee6a 100644
--- a/lib/cook-raw-quasi.js
+++ b/lib/cook-raw-quasi.js
@@ -1,10 +1,11 @@
 'use strict';
 
-function cookRawQuasi({transform}, raw) {
+function cookRawQuasi({transformSync}, raw) {
 	// This nasty hack is needed until https://github.com/babel/babel/issues/9242 is resolved.
 	const args = {raw};
 
-	transform('cooked`' + args.raw + '`', {
+	// Babel 8 removed synchronous `transform`; use `transformSync` instead.
+	transformSync('cooked`' + args.raw + '`', {
 		babelrc: false,
 		configFile: false,
 		plugins: [
