diff --git a/dist/shared/webpackbar.e03748fc.cjs b/dist/shared/webpackbar.e03748fc.cjs
index 4a2a28f34b079f6d331e252d4de8b7248ab0d09f..ea9a2b846c231fce8ddf28939bed126d334b386d 100644
--- a/dist/shared/webpackbar.e03748fc.cjs
+++ b/dist/shared/webpackbar.e03748fc.cjs
@@ -784,7 +784,7 @@ function eraseLines(count) {
 }
 
 const parseRequest = (requestStr) => {
-  const parts = (requestStr || "").split("!");
+  const parts = (typeof requestStr === "string" ? requestStr : "").split("!");
   const file = path__default.relative(
     process.cwd(),
     removeAfter("?", removeBefore(nodeModules, parts.pop()))
