diff --git a/polyfillLoaders/EventTarget.js b/polyfillLoaders/EventTarget.js
index 4e18ade7ba485849f17f28c94c42f0e0e01ac387..8f34f4f646c7f7becc208fb5a546c96034fc74dc 100644
--- a/polyfillLoaders/EventTarget.js
+++ b/polyfillLoaders/EventTarget.js
@@ -6,16 +6,15 @@
 let _ET;
 let ET;
 export default async function EventTarget() {
-    return ET || init();
+  return ET || init();
 }
 async function init() {
-    _ET = window.EventTarget;
-    try {
-        new _ET();
-    }
-    catch (_a) {
-        _ET = (await import('event-target-shim')).EventTarget;
-    }
-    return (ET = _ET);
+  _ET = window.EventTarget;
+  try {
+    new _ET();
+  } catch (_a) {
+    _ET = (await import("event-target-shim")).default.EventTarget;
+  }
+  return (ET = _ET);
 }
 //# sourceMappingURL=EventTarget.js.map